universal-login-react/src/ui/Migrating/EnterEmail.tsx

```diff
@@ -0,0 +1,77 @@
+import React, {useState} from 'react';
+import {ensureNotFalsy, isProperEmail} from '@unilogin/commons';
+import {WalletService} from '@unilogin/sdk';
+import {useClassFor, classForComponent} from '../utils/classFor';
+import {InputField} from '../commons/InputField';
+import {useInputField} from '../hooks/useInputField';
+import {PrimaryButton} from '../commons/Buttons/PrimaryButton';
+import '../styles/base/enterEmail.sass';
+import '../styles/themes/UniLogin/enterEmailThemeUniLogin.sass';
+import emailIcon from '../assets/icons/e-mail.svg';
+import {SecondaryButton} from '../commons/Buttons/SecondaryButton';
+import Spinner from '../commons/Spinner';
+import {OnboardingStepsWrapper} from '../Onboarding/OnboardingStepsWrapper';
+import {useHistory} from 'react-router';
+
+const emailValidator = {
+  validate: isProperEmail,
+  errorMessage: 'Email is not valid',
+};
+
+interface EnterEmailProps {
+  hideModal?: () => void;
+  walletService: WalletService;
+  onConfirm: (email: string) => Promise<void>;
+}
+
+export const EnterEmail = ({hideModal, walletService, onConfirm}: EnterEmailProps) => {
+  const history = useHistory();
+  const [email, setEmail, emailError] = useInputField([emailValidator]);
+  const [loading, setLoading] = useState(false);
+  const primaryButtonClassName = useClassFor('proceed-btn');
+
+  const onConfirmClick = async () => {
+    ensureNotFalsy(email, Error, 'Email missing');
+    setLoading(true);
+    await onConfirm(email);
```

Comment:
Is it possible that onConfirm callback will be not provided or won't change the modal state? If so, we could setLoading to false after calling onConfirm

Reply:
No, the component doesn't have sens if we are not changing screen